Framework/src/TaskRunner.cxx

```diff
@@ -398,17 +355,41 @@ void TaskRunner::finishCycle(DataAllocator& outputs)
   }
 }
 
-unsigned long TaskRunner::publish(DataAllocator& outputs)
+void TaskRunner::publishCycleStats()
+{
+  double cycleDuration = mTimerDurationCycle.getTime();
+  double rate = mNumberObjectsPublishedInCycle / (cycleDuration + mLastPublicationDuration);
+  double wholeRunRate = mTotalNumberObjectsPublished / mTimerTotalDurationActivity.getTime();
+  double totalDurationActivity = mTimerTotalDurationActivity.getTime();
+
+  // monitoring metrics
+  mCollector->send({ mNumberBlocks, "QC_task_Numberofblocks_in_cycle" });
+  mCollector->send({ cycleDuration, "QC_task_Module_cycle_duration" });
+  mCollector->send({ mLastPublicationDuration, "QC_task_Publication_duration" });
+  mCollector->send({ mNumberObjectsPublishedInCycle, "QC_task_Number_objects_published_in_cycle" });
+  mCollector->send({ rate, "QC_task_Rate_objects_published_per_second" });
+  mCollector->send({ mTotalNumberObjectsPublished, "QC_task_Total_objects_published_whole_run" });
+  mCollector->send({ totalDurationActivity, "QC_task_Total_duration_activity_whole_run" });
+  mCollector->send({ wholeRunRate, "QC_task_Rate_objects_published_per_second_whole_run" });
+}
+
+int TaskRunner::publish(DataAllocator& outputs)
 {
+  AliceO2::Common::Timer publicationDurationTimer;
+
+  TObjArray* array = mObjectsManager->getNonOwningArray();
+  int objectsPublished = array->GetEntries();
+
   auto concreteOutput = framework::DataSpecUtils::asConcreteDataMatcher(mMonitorObjectsSpec);
   outputs.adopt(
     Output{ concreteOutput.origin,
             concreteOutput.description,
             concreteOutput.subSpec,
             mMonitorObjectsSpec.lifetime },
-    dynamic_cast<TObject*>(mObjectsManager->getNonOwningArray()));
+    dynamic_cast<TObject*>(array));
 
-  return 1;
+  mLastPublicationDuration = publicationDurationTimer.getTime();
+  return objectsPublished;
```

Comment:
Also, note this change. Now we don't count all MonitorObjects as one object anymore (just because they were published in one TObjArray).